src/seraphis_main/contextual_enote_record_types.h

```diff
@@ -86,6 +86,79 @@ enum class SpEnoteSpentStatus : unsigned char
     SPENT_ONCHAIN
 };
 
+////
+// LegacyEnoteOriginContextV1
+// - info related to the transaction where a pre-RingCT enote was found
+// - note that an enote may originate off-chain in a partial tx where the tx id is unknown
+///
+struct LegacyEnoteOriginContextV1 final
+{
+    /// block index of tx (-1 if index is unknown)
+    std::uint64_t block_index{static_cast<std::uint64_t>(-1)};
+    /// timestamp of tx's block (-1 if timestamp is unknown)
+    std::uint64_t block_timestamp{static_cast<std::uint64_t>(-1)};
+    /// tx id of the tx (0 if tx is unknown)
+    rct::key transaction_id{rct::zero()};
+    /// index of the enote in the tx's output set (-1 if index is unknown)
+    std::uint64_t enote_tx_index{static_cast<std::uint16_t>(-1)};
+    /// index of the enote in the set [0, number of enotes in the chain with exact same amount) (-1 if index is unknown)
+    std::uint64_t enote_same_amount_ledger_index{static_cast<std::uint64_t>(-1)};
+    /// ledger index of the enote (-1 if index is unknown)
+    std::uint64_t enote_ledger_index{static_cast<std::uint64_t>(-1)};
```

Comment:
I think so, to quote jberman https://github.com/seraphis-migration/monero/pull/16#discussion_r1419847446
>This would only need to be removed if/when support for spending all of {pre-RingCT, coinbase, RingCT} outputs in the same ring is removed from the Seraphis lib

Reply:
That's only if the CLSAG struct inside txs uses the universal indexing method. If it uses (amount, same amount index) pairs, then we don't need both indices,